### PR TITLE
2024-09-09: Add weekly report

### PR DESCRIPTION
**Pull request by P-E-P**

### CODE REVIEW

**Comment by P-E-P** [comment]
> Not sure this counts... But we now have an upstream gcc build with some more checking (`--enable-checking=...`). See https://inbox.sourceware.org/buildbot/Zs8EYRjiXwSV2gq4@elastic.org/T/#t

Wdym ? Would you like this to be mentioned in the report ?

**Comment by dkm** [comment]
> > Not sure this counts... But we now have an upstream gcc build with some more checking (`--enable-checking=...`). See https://inbox.sourceware.org/buildbot/Zs8EYRjiXwSV2gq4@elastic.org/T/#t
> 
> What do you mean ? Would you like this to be mentioned in the report ?

These reports are read by many people and showing some activity in the upstream GCC project can be nice (and it can also make some publicity for sourceware infra). I know you are using scripts to gather everything that happens in the github world, but it's easy to miss some contribution upstream.

But feel free to ignore this if this is off topic, I don't have a strong opinion on the matter :)

**Comment by P-E-P** [comment]
You're right, we should talk about it.

File: 2024/2024-09-09-report.org [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+** Completed Activities
+
+- Strong types and `IndexVec` [[https://github.com/Rust-GCC/gccrs/pull/3122][PR3122]]
+- Avoid accidental insertion into map [[https://github.com/Rust-GCC/gccrs/pull/3155][PR3155]]
+- Add virtual destructor to AbstractExpr [[https://github.com/Rust-GCC/gccrs/pull/3147][PR3147]]
+- Fix v0-mangle1.rs test when run with ASAN [[https://github.com/Rust-GCC/gccrs/pull/3148][PR3148]]
+- Fix clang-format CI [[https://github.com/Rust-GCC/gccrs/pull/3152][PR3152]]
+- Asm generic il codegen [[https://github.com/Rust-GCC/gccrs/pull/3060][PR3060]]
+
```
Comment (dkm) [inline 2024/2024-09-09-report.org:20]:
```suggestion
- [[https://inbox.sourceware.org/buildbot/Zs8EYRjiXwSV2gq4@elastic.org/T/#t][Add support for building gccrs in upstream "all languages builder with extra checks" by sourceware's buildbots]]
```

File: 2024/2024-09-09-report.org [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+*** Contributors this week
+
+- [[https://github.com/braw-lee][Kushal Pal]]
+- [[https://github.com/badumbatish][Jasmine Tang]]
+- [[https://github.com/powerboat9][Owen Avery]]
+- [[https://github.com/P-E-P][Pierre-Emmanuel Patry]]
+
```
Comment (dkm) [inline 2024/2024-09-09-report.org:27]:
```suggestion
- [[https://github.com/dkm][Marc Poulhiès]]
```

**Review by dkm — COMMENTED** [review]
here's a small note on sourceware's bulidbot

**Comment by CohenArthur** [comment]
> Not sure this counts... But we now have an upstream gcc build with some more checking (`--enable-checking=...`). See https://inbox.sourceware.org/buildbot/Zs8EYRjiXwSV2gq4@elastic.org/T/#t

this is definitely something we need to talk about and something I had missed, thank you :+1:

**Pull request merged**